src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
+const DEFAULT_HEALTH = 100;
+const DAMAGE_VALUE = 50;
+
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = DEFAULT_HEALTH) {
+    this.health = health;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  removeDead() {
+    Animal.alive = Animal.alive.filter((item) => item.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(health, name) {
+    super(health, name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (herbivore instanceof Herbivore && !herbivore.hidden) {
+      herbivore.health -= DAMAGE_VALUE;
+    }
+
+    this.removeDead();
```

Comment:
[CODE PERFORMANCE] Avoid calling `removeDead` inside the `bite` method to prevent potential performance issues with nested loops. Instead, you should manage the removal of dead animals separately, perhaps in a routine that runs periodically or after a series of interactions.